awx/main/dispatch/worker/task.py

```diff
@@ -52,6 +54,8 @@ def run_callable(self, body):
         uuid = body.get('uuid', '<unknown>')
         args = body.get('args', [])
         kwargs = body.get('kwargs', {})
+        if 'correlation_id' in body:
+            GuidMiddleware.set_guid(body.pop('correlation_id'))
```

Comment:
When a dispatcher child process deserializes some message, if there's a GUID set, apply it to the current thread so all log messages in this process inherit it.